spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -370,14 +371,56 @@ public void alterNamespace(String[] namespace, NamespaceChange... changes) throw
 
   @Override
   public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException {
+    // Spark assumes that catalogs CASCADE by default. So if we have a namespace catalog
+    /// we need to attempt to recursively delete.
     if (asNamespaceCatalog != null) {
+      Namespace asNamespace = Namespace.of(namespace);
+      boolean exists = namespaceExists(namespace);
+
+      // TODO - Follow up
+      //
+      // If not exists, we can eagerly try to drop the namespace.
+      // If it throws, then that means that the user did not use IF EXISTS.
+      // If it does not throw, we can return false early as the user must have used IF EXISTS.
+
+      // Spark only throws the catalyst version of `NoSuchNamespaceException` if the namespace
+      // does not exist AND the user did not specify `IF EXISTS` in their query.
+      //
+      // If the namespace does not exist, but listNamespaces didn't throw an exception,
+      // we know the user used IF EXISTS and can return false early.
+      String[][] subNamespaces;
       try {
-        return asNamespaceCatalog.dropNamespace(Namespace.of(namespace));
+        subNamespaces = listNamespaces(namespace);
       } catch (org.apache.iceberg.exceptions.NoSuchNamespaceException e) {
         throw new NoSuchNamespaceException(namespace);
       }
-    }
 
+      if (!exists && (subNamespaces == null || subNamespaces.length == 0)) {
```

Comment:
If the `listNamespaces` call above didn't throw an exception then the namespace must exist. So this `if` is redundant because `!exists` is always going to be false.